### DISCUSSION TITLE
How can I listen to an entity colliding with other things?

### DISCUSSION THREAD

**Discussion by tristan-f-r** [Minestom/Minestom#756] [Help]
I would like to make a mob disappear whenever they hit anything.

**Comment by iam4722202468** [comment]
You can extend the entity class and override the tick function. Get the velocity before the tick is done, and calculate the expected position after the move `velocity / tps + current position - air resistance`. Compare this to the position after the tick, if it isn't the same position then you know the mob hit something and couldn't make the full move.
